From f90c344aec24b5b20fff5fa2a0c843c01c0ca375 Mon Sep 17 00:00:00 2001
From: Marc Mutz <marc.mutz@qt.io>
Date: Tue, 21 Oct 2025 17:53:06 +0200
Subject: [PATCH 504/553] QUnicodeTools: improve variable allocation in
 getSentenceBreaks()

Make 'pos' and 'prop' const, indicating that they're not modified by
the lengthy loop body, and don't re-use 'ucs4' and 'prop' from the
outer loop, make the inner loop have their own versions. This shadows
the outer loop ones, but -Wshadow is not in effect in implementation
files. I found it more important to avoid churn than to rename the
variables to avoid the shadowing. Shadowing is well-defined in C++.

These are in preparation of porting the function to QStringIterator.

Pick-to: 6.8 6.5
Change-Id: Ia8b136c2cf4c8bc70d7444456adae93aecf6138b
Reviewed-by: Ahmad Samir <a.samirh78@gmail.com>
(cherry picked from commit 961271052a8998d87ba84c0b56a5c55c8354c09e)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/corelib/text/qunicodetools.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/corelib/text/qunicodetools.cpp b/src/corelib/text/qunicodetools.cpp
index b339bb78a43..040b39e4621 100644
--- a/src/corelib/text/qunicodetools.cpp
+++ b/src/corelib/text/qunicodetools.cpp
@@ -407,7 +407,7 @@ static void getSentenceBreaks(const char16_t *string, qsizetype len, QCharAttrib
 {
     uchar state = SB::BAfter; // to meet SB1
     for (qsizetype i = 0; i != len; ++i) {
-        qsizetype pos = i;
+        const qsizetype pos = i;
         char32_t ucs4 = string[i];
         if (QChar::isHighSurrogate(ucs4) && i + 1 != len) {
             ushort low = string[i + 1];
@@ -417,7 +417,7 @@ static void getSentenceBreaks(const char16_t *string, qsizetype len, QCharAttrib
             }
         }
 
-        const QUnicodeTables::Properties *prop = QUnicodeTables::properties(ucs4);
+        const auto prop = QUnicodeTables::properties(ucs4);
         QUnicodeTables::SentenceBreakClass ncls = (QUnicodeTables::SentenceBreakClass) prop->sentenceBreakClass;
 
         Q_ASSERT(state <= SB::BAfter);
@@ -425,7 +425,7 @@ static void getSentenceBreaks(const char16_t *string, qsizetype len, QCharAttrib
         if (Q_UNLIKELY(state == SB::Lookup)) { // SB8
             state = SB::Break;
             for (qsizetype lookahead = i + 1; lookahead < len; ++lookahead) {
-                ucs4 = string[lookahead];
+                char32_t ucs4 = string[lookahead];
                 if (QChar::isHighSurrogate(ucs4) && lookahead + 1 != len) {
                     ushort low = string[lookahead + 1];
                     if (QChar::isLowSurrogate(low)) {
@@ -434,7 +434,7 @@ static void getSentenceBreaks(const char16_t *string, qsizetype len, QCharAttrib
                     }
                 }
 
-                prop = QUnicodeTables::properties(ucs4);
+                const auto prop = QUnicodeTables::properties(ucs4);
                 QUnicodeTables::SentenceBreakClass tcls = (QUnicodeTables::SentenceBreakClass) prop->sentenceBreakClass;
                 switch (tcls) {
                 case QUnicodeTables::SentenceBreak_Any:
-- 
2.51.2

